wayland-client/src/event_queue.rs

```diff
@@ -604,14 +604,16 @@ impl<State: 'static> QueueHandle<State> {
     /// This creates an implementation of [`ObjectData`] fitting for direct use with `wayland-backend` APIs
     /// that forwards all events to the event queue associated with this token, integrating the object into
     /// the [`Dispatch`]-based logic of `wayland-client`.
-    pub fn make_data<I: Proxy + 'static, U: Send + Sync + 'static>(
+    ///
+    /// Events will be dispatched via [`Dispatch`] to a `DelegateTo` generic type.
```

Comment:
I have a feeling that this is a bit too succinct, but I'm not quite sure how to expand it.

Reply:
Attempted to clarify that by adding a pseudocode example https://github.com/Smithay/wayland-rs/pull/699/commits/ee9140d38bbf6922653196a3fbe1200b175808cc#diff-ce00ed5e221ce826207c7b1cff410364eeb6e30c9322e7a875129b8a50871198R609-R627